app/controllers/items/history.js

```diff
@@ -0,0 +1,74 @@
+import detail from "./detail";
+import Ember from "ember";
+import _ from "lodash";
+
+export default detail.extend({
+  enableFooterActions: false,
+
+  groupedVersions() {
+    let versions = this.get("model.versions").toArray();
+
+    let results = _.map(versions, function(version) {
+      return _.omit(version.get("objectChanges"), [
+        "available_quantity",
+        "designated_quantity",
+        "on_hand_quantity",
+        "dispatched_quanitity",
+        "id",
+        "detail_type_id",
+        "created_at",
+        "received_at"
+      ]);
+    });
+    versions.forEach((version, index) =>
+      version.set("objectChanges", results[index])
+    );
+    return versions;
+  },
+
+  getGroupKey(isAction, action) {
+    const createdAt = moment(action.get("createdAt")).format("LL");
+    if (isAction) {
+      return `${action.get("user.fullName")}/${action.get(
+        "action"
+      )}/${createdAt}`;
+    } else {
+      return `${action.get("whodunnitName")}/${action.get(
+        "state"
+      )}/${createdAt}`;
+    }
+  },
```

Comment:
I'd suggest keeping the logic that checks the type of the record _inside_ the method. Makes the method "smarter" or "more magical" : )

```suggestion
  getGroupKey(change) {
    const isAction = change.get('constructor.modelName') === "item_action"; 
    const createdAt = moment(change.get("createdAt")).format("LL");
    const author = change.get(isAction ? "user.fullName" : "whodunnitName");
    const type = change.get(isAction ? "action" : "state");
    
    return `${author}/${type}/${createdAt}`;
  },
```